src/IK/tests/QPInverseKinematicsTest.cpp

```diff
@@ -0,0 +1,286 @@
+    const std::string controlledFrame = kinDyn->model().getFrameName(numberOfJoints);
+    out.endEffectorPose = toManifPose(kinDyn->getWorldTransform(controlledFrame));
+    out.joints.resize(jointsPos.size());
+    out.joints = iDynTree::toEigen(jointsPos);
+
+    return out;
+}
+
+System getSystem(int NrOfDOFs)
+{
+    System out;
+
+    // create the System
+    Eigen::Matrix<double, 6, 1> twist;
+    twist.setZero();
+    Eigen::VectorXd jointVelocity(NrOfDOFs);
+    jointVelocity.setZero();
+    Eigen::Matrix3d baseRotation;
+    baseRotation.setIdentity();
+    Eigen::Vector3d basePosition;
+    basePosition.setZero();
+    Eigen::VectorXd jointPosition(NrOfDOFs);
+    jointPosition.setZero();
+
+    out.dynamics = std::make_shared<FloatingBaseSystemKinematics>();
+    out.dynamics->setState({basePosition, baseRotation, jointPosition});
+
+    out.integrator = std::make_shared<ForwardEuler<FloatingBaseSystemKinematics>>(dT);
+    out.integrator->setDynamicalSystem(out.dynamics);
+
+    return out;
+}
+
+TEST_CASE("QP-IK")
+{
+    auto kinDyn = std::make_shared<iDynTree::KinDynComputations>();
+    auto parameterHandler = createParameterHandler();
+
+    constexpr double tolerance = 1e-2;
+
+    // set the velocity representation
+    REQUIRE(kinDyn->setFrameVelocityRepresentation(
+        iDynTree::FrameVelocityRepresentation::MIXED_REPRESENTATION));
+
+    for (std::size_t numberOfJoints = 20; numberOfJoints < 60; numberOfJoints += 15)
+    {
+        DYNAMIC_SECTION("Model with " << numberOfJoints << " joints")
+        {
+            // create the model
+            const iDynTree::Model model = iDynTree::getRandomModel(numberOfJoints);
+            REQUIRE(kinDyn->loadRobotModel(model));
+
+            const auto desiredSetPoints = getDesiredReference(kinDyn, numberOfJoints);
+
+            // Instantiate the handler
+            VariablesHandler variablesHandler;
+            variablesHandler.addVariable(robotVelocity, model.getNrOfDOFs() + 6);
+
+            auto system = getSystem(model.getNrOfDOFs());
+
+            // Set the frame name
+            const std::string controlledFrame = model.getFrameName(numberOfJoints);
+            parameterHandler->getGroup("SE3_TASK")
+                .lock()
+                ->setParameter("frame_name", controlledFrame);
+
+            // create the IK
+            auto ikAndTasks = createIK(parameterHandler, kinDyn, variablesHandler);
+
+            REQUIRE(ikAndTasks.se3Task->setSetPoint(desiredSetPoints.endEffectorPose,
+                                                    manif::SE3d::Tangent::Zero()));
+            REQUIRE(ikAndTasks.comTask->setSetPoint(desiredSetPoints.CoMPosition,
+                                                    Eigen::Vector3d::Zero()));
+            REQUIRE(ikAndTasks.regularizationTask->setSetPoint(desiredSetPoints.joints));
+
+            // propagate the inverse kinematics for
+            constexpr std::size_t iterations = 1200;
```

Comment:
This test required 20min to valgrind to be finished. Please, start from a point that would require less iterations to converge (like 10-15 maybe)

Reply:
Done fd6618167931b8f3ced0f1b2a0ec72fdb52c687a